python_modules/dagster/dagster/_core/definitions/automation_tick_evaluation_context.py

```diff
@@ -503,10 +503,10 @@ def _build_run_request_for_partition_key_range(
         ASSET_PARTITION_RANGE_START_TAG: partition_range_start,
         ASSET_PARTITION_RANGE_END_TAG: partition_range_end,
     }
-    partition_key = partition_range_start if partition_range_start == partition_range_end else None
```

Comment:
this was a bug that my test uncovered, we raise an exception at ASSET_MATERIALIZATION_PLANNED event time if you set both a range and a key. I opted to always set a range for backfill policies so that you don't have to have your code handle both

Reply:
seems like this was only an issue in tests and the actual asset backfill machinery just ignores run_request.partition_key